Src/FluentAssertions/Events/EventMonitor.cs

```diff
@@ -117,20 +120,67 @@ public void Dispose()
     {
         foreach (EventRecorder recorder in recorderMap.Values)
         {
-            recorder.Dispose();
+            DisposeSafeIfRequested(recorder);
         }
 
         recorderMap.Clear();
     }
 
+    private void DisposeSafeIfRequested(EventRecorder recorder)
+    {
+        if (options.ShouldIgnoreEventAccessorExceptions)
+        {
+            try
+            {
+                recorder.Dispose();
+            }
+            catch
+            {
+                // ignore
+            }
+        }
+        else
+        {
+            recorder.Dispose();
+        }
```

Comment:
```cs
try
{
    recorder.Dispose();
}
catch when (options.ShouldIgnoreEventAccessorExceptions)
{
    // ignore
}
```

Reply:
Wow, great improvement! Thanks! I'd never thought doing it with catch when.